velox/functions/prestosql/Split.cpp

```diff
@@ -0,0 +1,263 @@
+        });
+      } else {
+        const I limit = limits->valueAt<I>(0);
+        rows.applyToSelected([&](vector_size_t row) {
+          applyInner<true, I>(rawStrings[row], delim, limit, row, builder);
+        });
+      }
+
+      // Ensure that our result elements vector uses the same string buffer as
+      // the input vector of strings.
+      builder.setStringBuffers(
+          strings->base()->as<FlatVector<StringView>>()->stringBuffers());
+    } else if (
+        strings->isConstantMapping() and delims->isConstantMapping() and
+        (noLimit or limits->isConstantMapping())) {
+      // The next optimization case is for all constants.
+      const auto str = strings->valueAt<StringView>(0);
+      const auto delim = delims->valueAt<StringView>(0);
+
+      if (noLimit) {
+        const I limit = std::numeric_limits<I>::max();
+        applyInner<false, I>(str, delim, limit, 0, builder);
+      } else {
+        const I limit = limits->valueAt<I>(0);
+        applyInner<true, I>(str, delim, limit, 0, builder);
+      }
+
+      // Ensure that our result elements vector uses the same string buffer as
+      // the input vector of strings.
+      builder.setStringBuffers({strings->base()
+                                    ->as<ConstantVector<StringView>>()
+                                    ->getStringBuffer()});
+    } else {
+      // The rest of the cases are handled through this general path and no
+      // direct access.
+      applyDecoded<I>(rows, strings, delims, limits, builder);
+
+      // Ensure that our result elements vector uses the same string buffer as
+      // the input vector of strings.
+      builder.setStringBuffers(strings->base());
+    }
+
+    std::shared_ptr<ArrayVector> arrayVector =
+        std::move(builder).finish(context->pool());
+    context->moveOrCopyResult(arrayVector, rows, result);
+  }
+
+  template <typename I>
+  void applyDecoded(
+      const SelectivityVector& rows,
+      DecodedVector* strings,
+      DecodedVector* delims,
+      DecodedVector* limits,
+      ArrayBuilder<Varchar>& builder) const {
+    if (limits == nullptr) {
+      auto limit = std::numeric_limits<I>::max();
+      rows.applyToSelected([&](vector_size_t row) {
+        applyInner<false, I>(
+            strings->valueAt<StringView>(row),
+            delims->valueAt<StringView>(row),
+            limit,
+            row,
+            builder);
+      });
+    } else {
+      rows.applyToSelected([&](vector_size_t row) {
+        applyInner<true, I>(
+            strings->valueAt<StringView>(row),
+            delims->valueAt<StringView>(row),
+            limits->valueAt<I>(row),
+            row,
+            builder);
+      });
+    }
+  }
+
+  /**
+   * The inner most kernel of the vector operations for 'split'.
+   */
+  template <bool hasLimit = false, typename I>
+  inline void applyInner(
+      StringView input,
+      const StringView delim,
+      I limit,
```

Comment:
You don't need to template the type here since you are always passing a specific value. Just make it int32 or something.